pkg/capr/planner/certificaterotation_test.go

```diff
@@ -230,22 +230,36 @@ func Test_rotateCertificatesPlan(t *testing.T) {
 				tt.setup(mockPlanner)
 			}
 			controlPlane := createTestControlPlane(tt.version)
-			controlPlane.Spec.MachineGlobalConfig = *tt.machineGlobalConfig
+			if tt.machineGlobalConfig != nil {
+				controlPlane.Spec.MachineGlobalConfig = *tt.machineGlobalConfig
+			} else {
+				controlPlane.Spec.MachineGlobalConfig = rkev1.GenericMap{
+					Data: map[string]interface{}{},
+				}
+			}
+
 			controlPlane.Spec.ManagementClusterName = "somecluster"
 			if tt.rotateCertificates != nil {
 				controlPlane.Spec.RotateCertificates = tt.rotateCertificates
+			} else {
+				controlPlane.Spec.RotateCertificates = &rkev1.RotateCertificates{}
 			}
 			entry := createTestPlanEntry(capr.DefaultMachineOS)
 			if tt.entryIsControlPlane {
 				entry.Machine.Labels[capr.ControlPlaneRoleLabel] = "true"
 				entry.Metadata.Labels[capr.ControlPlaneRoleLabel] = "true"
+			} else {
+				// to avoid implausible join server error
+				entry.Metadata.Annotations = map[string]string{
+					capr.JoinedToAnnotation: controlPlane.GetName(),
```

Comment:
@Oats87 correct me if I'm wrong but this is not what `joined-to` is intended to be used for. This annotation is meant to specify which server was joined to, not which controlplane object the node is a part of. I'm curious as to why this was needed.

Reply:
agree, am curious as to why this is necessary

Reply:
Sorry for the confusion....my mistake....just realised it should have been set to expected joined server... @rayandas  has made the change....this annotation is required for worker only tests...https://github.com/rancher/rancher/blob/331d18b352cfd58d5e0fc4815dffad97ca30eb49/pkg/capr/planner/config.go#L110